File: animationjava/src/main/java/com/codekul/animationjava/MainActivity.java
package com.codekul.animationjava;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.btnrotate).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                animate(R.anim.rotate);
            }
        });
        findViewById(R.id.btnzoom).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                animate(R.anim.zoom);
            }
        });

        findViewById(R.id.btnmixed).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                animate(R.anim.mixed);
            }
        });


    }
    private void animate(int res){
    findViewById(R.id.imageView).startAnimation(android.view.animation.AnimationUtils.loadAnimation(this,res));
    }
}

File: compoundlayouttryKotlin/src/main/java/com/codekul/compoundlayouttryjava/MainActivity.kt
package com.codekul.compoundlayouttryjava

import android.content.Context
import android.media.Image
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.zip.Inflater

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnShow.setOnClickListener {
            var vw :View = layoutInflater.inflate(
                    R.layout.compound_layer,
                    null,
                    false
            )
            vw.findViewById<ImageView>(R.id.img).setOnClickListener {
                (it as ImageView).setImageResource( R.drawable.ic_error_black_24dp)
            }
layoutroot.addView(vw)
        }
    }
}
